File: Example/deeplinkingMobile/WalletConnect.js
// App.js
import DIAMWalletConnectionSDK from 'diamwallet-sdk';
import React, {useState, useEffect} from 'react';
import {View, Text, Button, StyleSheet, Alert, TextInput} from 'react-native';

function WalletConnet() {
  const [connectionStatus, setConnectionStatus] = useState(false);
  const [address, setAddress] = useState('');
  const [balance, setBalance] = useState(null);
  const [sendView, setSendView] = useState(false);
  const [signView, setSignView] = useState(false);
  const [amount, setAmount] = useState('');
  const [publicAddress, setPublicAddress] = useState('');
  const [validPublicAddress, setValicPublicAddress] = useState(null);
  const rnDiamWallet = new DIAMWalletConnectionSDK({
    platform: 'mobile',
    appCallback: 'app-scheme://callback',
    appName: 'Demo App',
    network: 'testnet', //"mainnet"
  });
  const amountRejex = /^\d*\.?\d{0,6}$/;

  const [selectedXDR, setSelectedXdr] = useState('');

  const getDIAMBalance = async () => {
    let balanceResponse = await rnDiamWallet.getBalance();
    setBalance(balanceResponse?.data?.balance || '0');
  };

  const connectWallet = async () => {
    try {
      console.log('Connecting wallet...');
      const result = await rnDiamWallet.connectWallet();
      console.log('Wallet connected!', result);
      if (result.status === true) {
        setAddress(result.address);
        setConnectionStatus(result.status);
      } else {
        Alert.alert('Errors', result.data, [{text: 'OK'}]);
      }
    } catch (error) {
      console.error('Failed to connect wallet:', error);
      setConnectionStatus(false);
      Alert.alert(`Failed to connect: \n${error.message}`);
    } finally {
      let balanceResponse = await rnDiamWallet.getBalance();
      console.log(balanceResponse);
      setBalance(balanceResponse?.data?.balance || '0');
      // sdk.disconnect();
    }
  };

  const sendTransaction = async () => {
    let transactionData = {
      amount: amount,
      toAddress: publicAddress,
      signTransaction: false,
    };
    try {
      const result = await rnDiamWallet.sendTransaction(transactionData);
      console.log('Send Initiated!---', result.success);
      if (result.success === true) {
        setAmount('');
        setPublicAddress('');
        setSendView(false);
        setValicPublicAddress(null);
        Alert.alert(
          'Transaction Success',
          `Hash: ${result.transactionDetails.hash}`,
          [{text: 'OK'}],
        );
      } else {
        Alert.alert('Transaction Failed', result.transactionStatus, [
          {text: 'OK'},
        ]);
      }
      // setAddress(result.address);
      // setConnectionStatus(result.status);
      // getDIAMBalance();
    } catch (error) {
      console.error('Failed to send transaction:', error);
      setAmount('');
      setPublicAddress('');
      setSendView(false);
      setValicPublicAddress(null);
    } finally {
    }
  };

  const handleDisconnect = () => {
    rnDiamWallet.disconnect();
    setAddress('');
    setConnectionStatus(false);
    setBalance(null);
    setSignView(false);
    setSendView(false);

    setAmount('');
    setPublicAddress('');
    setValicPublicAddress(null);
  };
  const initialzile = async () => {
    let data = await rnDiamWallet.initializeSdk();
    if (data) {
      setAddress(data.address);
      setConnectionStatus(data.status);
      let balanceResponse = await rnDiamWallet.getBalance();
      setBalance(balanceResponse.data.balance);
    }
  };
  useEffect(() => {
    initialzile();
  }, []);

  const addressValidation = async () => {
    let valid = await rnDiamWallet.validatePublicAddress(publicAddress);
    console.log(valid);
    if (valid.valid === false) {
      Alert.alert(
        'Address Validation',
        `${publicAddress} \n\nis not valid address`,
      );
      return;
    }
    setValicPublicAddress(valid.valid);
  };

  const signTransaction = async () => {
    let transactionData = {
      amount: amount,
      toAddress: publicAddress,
      signTransaction: true,
    };
    try {
      const result = await rnDiamWallet.signTransaction(transactionData);
      if (result.success === true) {
        setSelectedXdr(result.signedXdr);
      } else {
        Alert.alert('Transaction Failed', result.transactionStatus, [
          {text: 'OK'},
        ]);
      }
      // setAddress(result.address);
      // setConnectionStatus(result.status);
      // getDIAMBalance();
    } catch (error) {
      console.error('Failed to send transaction:', error);
    } finally {
    }
  };

  const sendSignedXDRTransaction = async () => {
    let transactionData = {
      // amount: '1',
      // toAddress: 'GBALA3OP3SB5CWR2CZRBKJXECBDBHNFTYJEN7JHIX4XYFC427CJ2DAUT',
      signTransaction: false,
      xdr: selectedXDR,
    };
    try {
      const result = await rnDiamWallet.sendTransaction(transactionData);
      console.log('Send Initiated!---', result.success);
      if (result.success === true) {
        setSelectedXdr('');
        Alert.alert(
          'Transaction Success',
          `Hash: ${result.transactionDetails.hash}`,
          [{text: 'OK'}],
        );
        setAmount('');
        setPublicAddress('');
        setSignView(false);
        setValicPublicAddress(null);
      } else {
        Alert.alert('Transaction Failed', result.transactionStatus, [
          {text: 'OK'},
        ]);
      }
      // setAddress(result.address);
      // setConnectionStatus(result.status);
      // getDIAMBalance();
    } catch (error) {
      console.error('Failed to send transaction:', error);
      setAmount('');
      setPublicAddress('');
      setSignView(false);
      setValicPublicAddress(null);
    } finally {
    }
  };

  return (
    <View style={styles.container}>
      <Text style={styles.status}>{connectionStatus}</Text>
      {address !== '' && (
        <Text style={styles.balance}>
          Wallet Address:{' '}
          {'\n' + address.slice(0, 7) + '...' + address.slice(-7)}
        </Text>
      )}
      {balance !== null && (
        <Text style={styles.balance}>Available Balance: {balance}</Text>
      )}
      <View style={styles.buttonContainer}>
        {connectionStatus === false ? (
          <Button
            title="Connect Wallet"
            onPress={connectWallet}
            // disabled={sdk.isConnected()}
          />
        ) : (
          <>
            <View style={{marginVertical: 10}}>
              <Button
                title="Get Balance"
                onPress={getDIAMBalance}

                // disabled={sdk.is()}
              />
            </View>
            <View
              style={{
                marginVertical: 10,
                flexDirection: 'row',
                alignItems: 'center',
                justifyContent: 'space-between',
                // gap: 4,
                width: '100%',
              }}>
              <View style={{width: `${100 / 2.2}%`}}>
                <Button
                  title="Sign Transaction"
                  onPress={() => {
                    setSignView(true);
                    setAmount('');
                    setPublicAddress('');
                    setValicPublicAddress(null);
                  }}
                  // style={{width: '48%'}}

                  // disabled={sdk.is()}
                />
              </View>
              <View style={{width: `${100 / 2.2}%`}}>
                <Button
                  title="Send Transaction"
                  onPress={
                    // sendTransaction
                    () => {
                      setSendView(true);
                      setAmount('');
                      setPublicAddress('');
                      setValicPublicAddress(null);
                    }
                  }
                  // style={{width: '48%'}}

                  // disabled={sdk.is()}
                />
              </View>
            </View>

            {sendView === true && (
              <>
                <View
                  style={{
                    width: '100%',
                    borderWidth: 1,
                    alignItems: 'center',
                  }}>
                  <TextInput
                    style={{
                      paddingVertical: 2,
                      width: '100%',
                      paddingHorizontal: 10,
                      color: '#000',
                    }}
                    placeholder="Enter Public Address"
                    value={publicAddress}
                    keyboardType="email-address"
                    onChangeText={e => {
                      setPublicAddress(e);
                    }}
                  />
                </View>
                <View
                  style={{
                    width: '100%',
                    borderWidth: 1,
                    alignItems: 'center',
                  }}>
                  <TextInput
                    style={{
                      paddingVertical: 2,
                      width: '100%',
                      paddingHorizontal: 10,
                      color: '#000',
                    }}
                    returnKeyType="done"
                    placeholder="Enter amount"
                    value={amount}
                    keyboardType="number-pad"
                    onChangeText={e => {
                      if (e === '' || amountRejex.test(e)) {
                        setAmount(e);
                      }
                    }}
                  />
                </View>
                <View
                  style={{
                    width: '100%',
                    flexDirection: 'row',
                    justifyContent: 'space-between',
                    marginTop: 10,
                  }}>
                  <View style={{width: `30%`}}>
                    <Button
                      title="Validate address"
                      onPress={addressValidation}
                      disabled={
                        publicAddress === '' || publicAddress.length < 5
                      }
                    />
                  </View>
                  <View style={{width: '30%'}}>
                    <Button
                      title="Send"
                      onPress={sendTransaction}
                      disabled={
                        amount === '' ||
                        validPublicAddress === null ||
                        validPublicAddress === false
                      }
                    />
                  </View>
                  <View style={{width: '30%'}}>
                    <Button
                      title="Cancel"
                      onPress={() => {
                        setAmount('');
                        setPublicAddress('');
                        setSendView(false);
                        setValicPublicAddress(null);
                      }}
                    />
                  </View>
                </View>
              </>
            )}
            {signView === true && (
              <>
                <View
                  style={{
                    width: '100%',
                    borderWidth: 1,
                    alignItems: 'center',
                  }}>
                  <TextInput
                    style={{
                      paddingVertical: 2,
                      width: '100%',
                      paddingHorizontal: 10,
                      color: '#000',
                    }}
                    returnKeyType="done"
                    editable={selectedXDR === ''}
                    placeholder="Enter Public Address"
                    value={publicAddress}
                    keyboardType="email-address"
                    onChangeText={e => {
                      setPublicAddress(e);
                      setValicPublicAddress(null);
                    }}
                  />
                </View>
                <View
                  style={{
                    width: '100%',
                    borderWidth: 1,
                    alignItems: 'center',
                  }}>
                  <TextInput
                    style={{
                      paddingVertical: 2,
                      width: '100%',
                      paddingHorizontal: 10,
                      color: '#000',
                    }}
                    returnKeyType="done"
                    editable={selectedXDR === ''}
                    placeholder="Enter amount"
                    value={amount}
                    keyboardType="number-pad"
                    onChangeText={e => {
                      if (e === '' || amountRejex.test(e)) {
                        setAmount(e);
                      }
                    }}
                  />
                </View>
                <View
                  style={{
                    width: '100%',
                    flexDirection: 'row',
                    justifyContent: 'space-between',
                    marginTop: 10,
                  }}>
                  <View style={{width: `30%`}}>
                    <Button
                      title="Validate address"
                      onPress={addressValidation}
                      disabled={
                        publicAddress === '' ||
                        publicAddress.length < 5 ||
                        selectedXDR !== ''
                      }
                    />
                  </View>
                  <View style={{width: '30%'}}>
                    <Button
                      title="Sign"
                      onPress={signTransaction}
                      disabled={
                        amount === '' ||
                        validPublicAddress === null ||
                        validPublicAddress === false ||
                        selectedXDR !== ''
                      }
                    />
                  </View>
                  <View style={{width: '30%'}}>
                    <Button
                      title="Cancel"
                      onPress={() => {
                        setAmount('');
                        setPublicAddress('');
                        setSignView(false);
                        setValicPublicAddress(null);
                      }}
                    />
                  </View>
                </View>
              </>
            )}

            {selectedXDR !== '' && (
              <View style={{marginVertical: 10}}>
                <Button
                  title="Send Signed Transaction "
                  onPress={sendSignedXDRTransaction}
                  // disabled={sdk.is()}
                />
              </View>
            )}

            <View style={{marginVertical: 10}}>
              <Button
                title="Disconnect Wallet"
                onPress={handleDisconnect}
                // disabled={sdk.is()}
              />
            </View>
          </>
        )}
      </View>
    </View>
  );
}

export default WalletConnet;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 20,
  },
  status: {
    fontSize: 18,
    marginBottom: 20,
  },
  balance: {
    fontSize: 16,
    // marginBottom: 20,
    textAlign: 'center',
  },
  buttonContainer: {
    width: '100%',
  },
});

File: Example/flutter_example/lib/screens/wallet_connect.dart
import 'package:diamwallet_sdk_flutter/diamwallet_sdk_flutter.dart';
import 'package:flutter/material.dart';

class WalletConnectScreen extends StatefulWidget {
  const WalletConnectScreen({super.key});

  @override
  _WalletConnectScreenState createState() => _WalletConnectScreenState();
}

class _WalletConnectScreenState extends State<WalletConnectScreen> {
  TextEditingController toAddressController = TextEditingController();
  TextEditingController amountController = TextEditingController();
  bool isConnected = false;
  String walletAddress = '';
  String balance = '';
  String amount = '';
  String toAddress = '';
  bool sendView = false;
  bool sendBep20View = false;
  String signedXdr = '';

  final diamWalletSdk = DIAMWalletConnectionSDK(
    options: {
      "platform": 'mobile',
      "appCallback": 'app-scheme://callback',
      "appName": 'DIAM Wallet SDK Demo',
      "network": 'testnet', //"mainnet"
    },
  );
  @override
  void initState() {
    super.initState();
    initializeWallet();
  }

  void initializeWallet() async {
    // Assume your SDK has an init method
    final result = await diamWalletSdk.initializeSdk();
    if (result?['status'] == true) {
      setState(() {
        isConnected = true;
        walletAddress = result?['address'];
      });
      getBalance();
    }
  }

  void connectWallet() async {
    try {
      final result = await diamWalletSdk.connectWallet();
      if (result['status'] == true) {
        setState(() {
          isConnected = true;
          walletAddress = result['address'];
        });
        getBalance();
      } else {
        showAlert('Connection Failed', result['data']);
      }
    } catch (e) {
      showAlert('Error', e.toString());
    }
  }

  void getBalance() async {
    final res = await diamWalletSdk.getBalance();

    print(res);
    setState(() {
      balance = res['data']?['balance'] ?? '0';
    });
  }

  void sendTransaction({required bool isBep20}) async {
    print(toAddress);
    final valid =
        isBep20
            ? await diamWalletSdk.validateBep20PublicAddress(
              toAddress,
              'DIAM (BEP20)',
            )
            : await diamWalletSdk.validatePublicAddress(toAddress);
    print(valid);
    if (!valid['valid']!) {
      showAlert('Invalid Address', '$toAddress is not valid');
      return;
    }

    final txData = {
      'amount': amount,
      'toAddress': toAddress,
      'signTransaction': false,
    };

    final result =
        isBep20
            ? await diamWalletSdk.sendBEP20Transaction(txData)
            : await diamWalletSdk.sendTransaction(txData);

    if (result['success'] == true) {
      showAlert('Success', 'Hash: ${result['transactionDetails']['hash']}');
      setState(() {
        sendView = false;
        sendBep20View = false;
        amount = '';
        toAddress = '';
      });
    } else {
      showAlert('Failed', result['transactionStatus']);
    }
  }

  void showAlert(String title, String message) {
    showDialog(
      context: context,
      builder:
          (_) => AlertDialog(
            title: Text(title),
            content: Text(message),
            actions: [
              TextButton(
                onPressed: () => Navigator.pop(context),
                child: Text('OK'),
              ),
            ],
          ),
    );
  }

  Widget transactionInputForm({required bool isBep20}) {
    return Container(
      padding: EdgeInsets.all(10),
      margin: EdgeInsets.symmetric(vertical: 10),
      decoration: BoxDecoration(
        borderRadius: BorderRadius.circular(10),

        border: Border.all(width: 1, color: Colors.grey),
      ),
      child: Column(
        children: [
          Container(
            padding: EdgeInsets.zero,
            width: double.infinity,

            child: Text(
              isBep20 ? "Send DIAM BEP20" : "Send DIAM",
              style: TextStyle(fontWeight: FontWeight.w700, fontSize: 18),
            ),
          ),
          TextField(
            controller: toAddressController,
            decoration: InputDecoration(labelText: 'Public Address'),
            onChanged: (value) => toAddress = value,
          ),
          TextField(
            controller: amountController,
            decoration: InputDecoration(labelText: 'Amount'),
            keyboardType: TextInputType.number,
            onChanged: (value) => amount = value,
          ),
          SizedBox(height: 10),
          Container(
            padding: EdgeInsets.all(0),
            width: double.infinity,
            child: Center(
              child: Row(
                spacing: 20,
                mainAxisAlignment: MainAxisAlignment.center,
                crossAxisAlignment: CrossAxisAlignment.center,
                children: [
                  ElevatedButton(
                    onPressed: () => sendTransaction(isBep20: isBep20),
                    child: Text("Submit"),
                  ),
                  ElevatedButton(
                    onPressed:
                        () => setState(() {
                          sendView = false;
                          sendBep20View = false;
                          amount = '';
                          toAddress = '';
                          amountController.clear();
                          toAddressController.clear();
                        }),
                    child: Text("Cancel"),
                  ),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }

  void disconnectWallet() {
    setState(() {
      isConnected = false;
      walletAddress = '';
      balance = '';
      amount = '';
      toAddress = '';
      sendView = false;
      sendBep20View = false;
    });
    diamWalletSdk.disconnect();
    showAlert('Disconnected', 'Wallet has been disconnected.');
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Padding(
        padding: EdgeInsets.all(16.0),
        child: Center(
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            mainAxisAlignment: MainAxisAlignment.center,
            children: [
              Container(
                width: double.infinity,
                margin: EdgeInsets.only(bottom: 40),
                child: Text(
                  'DIAM Wallet SDK Demo',
                  style: TextStyle(fontSize: 20, fontWeight: FontWeight.w700),
                  textAlign: TextAlign.center,
                ),
              ),
              Text(
                'Connection Status: ${isConnected ? "Connected" : "Disconnected"}',
              ),
              if (walletAddress.isNotEmpty)
                Text(
                  'Wallet Address: ${walletAddress.substring(0, 7)}...${walletAddress.substring(walletAddress.length - 7)}',
                ),
              if (balance.isNotEmpty) Text('Balance: $balance DIAM'),
              SizedBox(height: 20),

              isConnected
                  ? Expanded(
                    child: SingleChildScrollView(
                      child: Column(
                        children: [
                          ElevatedButton(
                            style: ElevatedButton.styleFrom(
                              minimumSize: Size(200, 40),
                            ),
                            onPressed: getBalance,
                            child: Text('Get Balance'),
                          ),
                          ElevatedButton(
                            style: ElevatedButton.styleFrom(
                              minimumSize: Size(200, 40),
                            ),
                            onPressed: () {
                              setState(() {
                                sendView = true;
                                sendBep20View = false;
                                toAddress = "";
                                amount = "";
                                amountController.clear();
                                toAddressController.clear();
                              });
                            },
                            child: Text('Send DIAM'),
                          ),
                          ElevatedButton(
                            onPressed: () {
                              setState(() {
                                sendBep20View = true;
                                sendView = false;
                                toAddress = "";
                                amount = "";
                                amountController.clear();
                                toAddressController.clear();
                              });
                            },
                            style: ElevatedButton.styleFrom(
                              minimumSize: Size(200, 40),
                            ),
                            child: Text('Send DIAM (BEP20)'),
                          ),
                          if (sendView) transactionInputForm(isBep20: false),
                          if (sendBep20View)
                            transactionInputForm(isBep20: true),

                          ElevatedButton(
                            onPressed: disconnectWallet,
                            style: ElevatedButton.styleFrom(
                              // backgroundColor: Colors.redAccent,
                              minimumSize: Size(200, 40),
                            ),
                            child: Text('Disconnect Wallet'),
                          ),
                        ],
                      ),
                    ),
                  )
                  : Container(
                    padding: EdgeInsets.all(0),
                    child: ElevatedButton(
                      onPressed: connectWallet,
                      child: Text('Connect Wallet'),
                    ),
                  ),
            ],
          ),
        ),
      ),
    );
  }
}
